src/regex/Regex.java

```diff
@@ -1,46 +1,93 @@
+        for (String word: TEKmentors) {
+        	Pattern pattern = Pattern.compile("\\w+");
+        	Matcher matcher = pattern.matcher(word);
+        	if(matcher.find()) {
+        		word = matcher.group();
+        		System.out.println(word);
+        	}
+        	
+        }
+
+
+        // 3. Find all the occurences of any form of 'book' in the following paragraph.
+        //use regex to match the occurences and store the count of books in an int.
         
         int bookCount = 0;
         String bookText = "Books are the keys to knowledge.  I didn't like to read books as a child, but in college I started enjoying learning and reading books. You can borrows books from the library, or you can buy them from the bookstore. I'm not sure if I prefer paperback books or hardcover books.  I'm such a nerd that I even like textbooks.  With the advent of technology, you can even buy digital books, such as kindle-books, nook-books, or other e-books. My personal favorite book format are pdf-books, because I don't have to carry so many books around wherever I go.  All the books are on my ipad or laptop.  When I lived abroad, they would give books to students absolutely free.  Free books for a student of knowledge is like a kid in a candy store.  So wipe the dust off of your books, and remember the slogan from 'reading rainbow' : 'Take a look! It's in a book! Reading Rainbow!";
         System.out.println(bookText);
+        
+        Pattern pattern1 = Pattern.compile("book|Book|books|Books");
+        Matcher matcher1 = pattern1.matcher(bookText);
+        while(matcher1.find()) {
+        	bookCount++;
+        }
+        System.out.println(bookCount);
 
-        // 4.a Create an array of all the words besides the word 'sleepy'.  Each word does not have to be a separate element, although you can split it that way if you wish.  We just want an array that everything that is not 'sleepy'.  
+        // 4.a Create an array of all the words besides the word 'sleepy'.
+        //Each word does not have to be a separate element, although you can split it that way if you wish.
+        //We just want an array that everything that is not 'sleepy'.  
 
         String sleepy = "I felt sleepy because I saw the others were sleepy and because I knew I should feel sleepy, but I wasn't really sleepy.  If you're sleepy and you know it, clap your hands.  Keep on being sleepy until you actually become sleepy";
-
+        
+        
+        String[] awake = sleepy.split("sleepy");
+        System.out.println(Arrays.toString(awake));
+        
         // 4.b combine the array that you just created into a string
         String not_sleepy; //punctuation marks will be here
-
+        	not_sleepy = String.join(" ", awake);
+        	System.out.println(not_sleepy);
         //4.c remove the punctuation marks from the notSleepy string.
         String notSleepy; //no punctuation marks should be here.
-
+        	notSleepy = not_sleepy.replaceAll("[^a-zA-Z]", " ");
```

Comment:
Rather than _**removing**_ the punctuation, the punctuation is being replaced with a space character, adding yet more white-space to the String